{@
    from "lib/std/error.pad" import Error

    def positive(n):
        if type(n) != "int":
            return nil, Error("argument is not int", Error.TYPE)
        end

        if n >= 0:
            return nil, Error("value is not negative", Error.VALUE)
        end

        return -n, nil
    end

    n, err = positive(nil)
    if err != nil:
        puts(err.what())
    end

    n, err = positive(1)
    if err != nil:
        puts(err.what())
    end

    n, _ = positive(-1)
    puts(n)
@}